pkg/controller/jobs/pod/pod_controller.go

```diff
@@ -934,7 +965,11 @@ func (p *Pod) ConstructComposableWorkload(ctx context.Context, c client.Client,
 		if err := controllerutil.SetControllerReference(object, wl, c.Scheme()); err != nil {
 			return nil, err
 		}
-
+		labelsToCopy, err := p.getWorkloadLabels(labelKeysToCopy)
```

Comment:
This code looks repeated for groups and non groups., but the function already makes the distinction inside. I would suggest to call it early once, and pass the labels when creating the workload object already. This would also match the flow for other job CRDs, and no need to merge.

Reply:
Your idea would make it cleaner but here is the reason why its proposed this way:

In the case of pod group we select only the active pods and cleanup excess pods 
````go
// Cleanup extra pods if there's any
	if excessPodsCount := len(activePods) - groupTotalCount; excessPodsCount > 0 {
		sortActivePods(activePods)
		err = p.removeExcessPods(ctx, c, r, activePods[len(activePods)-excessPodsCount:])
		if err != nil {
			return nil, err
		}
		p.list.Items = activePods[:len(activePods)-excessPodsCount]
	}
````
And this happens later in the code (after the workload object is created). So at the moment of workload creation some excess pods potentially could still be there. To be safe, I don't want to copy the labels from the excess pods so I should to my copying after the cleanup

Reply:
Ah, got it :) So this is to handle a situation when there is a mismatch, but in one of the excess pods.